### PR TITLE
fix: update to React Native 0.73.4 and Expo 50

### PR DESCRIPTION
**Pull request by paulschreiber**
## Description
Update to React Native 0.73.4 and Expo 50. Replaces #735 and #722.

### CODE REVIEW

**Review by shrouxm — APPROVED** [review]
i have many non-blocking questions, but looks good!

File: .github/workflows/build-shared-ios.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ on:
 jobs:
   ios-build:
     name: iOS Build
-    runs-on: macos-latest
+    runs-on: macos-14
```
[the remark this answers is not in the document]
Comment (paulschreiber) [reply .github/workflows/build-shared-ios.yml:43]: GitHub is still aliasing -latest to macOS 12. They are skipping aliasing it macOS 13 and will alias it to macOS 14 later this year: https://github.blog/changelog/2024-01-30-github-actions-macos-14-sonoma-is-now-available/

File: dev-client/app.json [hunk 2]
```diff
@@ -1,4 +1,3 @@
 {
   "name": "Terraso LandPKS",
```
[the remark this answers is not in the document]
Comment (paulschreiber) [reply dev-client/app.json:2]: The user-visible app name should have a space.

File: dev-client/ios/TerrasoLandPKS/Info.plist [hunk 3]
```diff
@@ -44,12 +44,18 @@
 	<dict>
 		<key>NSExceptionDomains</key>
 		<dict>
-			<key>localhost</key>
+			<key>exp.direct</key>
```
[the remark this answers is not in the document]
Comment (paulschreiber) [reply dev-client/ios/TerrasoLandPKS/Info.plist:47]: Not sure. But expo seems to use this for local development.

File: dev-client/ios/Podfile [hunk 4]
```diff
@@ -39,22 +42,14 @@ target 'TerrasoLandPKS' do
...
-    # Upcoming versions of React Native may rely on get_default_flags(), but
-    # we make it explicit here to aid in the React Native upgrade process.
-    :hermes_enabled => flags[:hermes_enabled],
-    :fabric_enabled => flags[:fabric_enabled],
     # Enables Flipper.
     #
     # Note that if you have use_frameworks! enabled, Flipper will not work and
     # you should disable the next line.
-    #:flipper_configuration => flipper_config,
+    :flipper_configuration => flipper_config,
```
[the remark this answers is not in the document]
Comment (paulschreiber) [reply dev-client/ios/Podfile]: Sounds good. I've removed the flipper references from both iOS and Android.

File: dev-client/android/build.gradle [hunk 5]
```diff
@@ -2,27 +2,28 @@
...
-        buildToolsVersion = "33.0.0"
+        buildToolsVersion = "34.0.0"
         minSdkVersion = 21
-        compileSdkVersion = 33
-        targetSdkVersion = 33
-        kotlinVersion = findProperty('android.kotlinVersion') ?: '1.8.10'
+        compileSdkVersion = 34
+        targetSdkVersion = 34
+        ndkVersion = "25.1.8937393"
+        kotlinVersion = "1.8.0"
```
[the remark this answers is not in the document]
Comment (paulschreiber) [reply dev-client/android/build.gradle]: Interesting.
* I took 1.8.0 from the React Native Upgrade Helper: https://react-native-community.github.io/upgrade-helper/?from=0.72.10&to=0.73.4
* the Expo upgrade helper uses 1.8.10: https://docs.expo.dev/bare/upgrade/?fromSdk=49&toSdk=50

I updated to `1.9.22`.

File: dev-client/android/app/src/main/java/org/terraso/landpks/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.fabricEnabled
+import com.facebook.react.defaults.DefaultReactActivityDelegate
+
+class MainActivity : ReactActivity() {
+
+  /**
+   * Returns the name of the main component registered from JavaScript. This is used to schedule
+   * rendering of the component.
+   */
+  override fun getMainComponentName(): String = "Terraso LandPKS"
```
[the remark this answers is not in the document]
Comment (paulschreiber) [reply dev-client/android/app/src/main/java/org/terraso/landpks/MainActivity.kt:15]: Not sure. Happy to change it back later if needed.

**Pull request merged**